src/components/Table.tsx

```diff
@@ -0,0 +1,177 @@
+import get = require('lodash/get');
+import isPlainObject = require('lodash/isPlainObject');
+import map = require('lodash/map');
+import reverse = require('lodash/reverse');
+import sortBy = require('lodash/sortBy');
+import * as React from 'react';
+import FaSort = require('react-icons/lib/fa/sort');
+import styled from 'styled-components';
+
+import Button from './Button';
+import { Flex } from './Grid';
+import theme from '../theme';
+
+type TableDataItem =
+	| void
+	| string
+	| JSX.Element
+	| {
+			value: any;
+			display: string | JSX.Element;
+		};
+
+export interface TableDataRow {
+	[key: string]: TableDataItem;
+}
+
+type TableSortFunction = <T>(a: T, b: T) => number;
+
+interface TableHeaderItem {
+	field: string;
+	icon?: string;
+	label?: string | JSX.Element;
+	sortable?: boolean | TableSortFunction;
```

Comment:
I would also love to be able to pass a `render()` parameter to control how each field is presented!
Probably in a separate PR.